sync-core/src/test/java/com/cloudant/sync/replication/BasicPullStrategyMockTest.java

```diff
@@ -147,6 +164,58 @@ public void call_coordinationThreadInterrupted_replicatorStopCorrectly() throws
         verify(mockListener).complete(any(ReplicationStrategyCompleted.class));
         verify(mockListener, never()).error(any(ReplicationStrategyErrored.class));
     }
+
+    @Test
+    public void testReplicationDocWithEmptyId() throws Exception {
+        CouchDB mockRemoteDb = mock(CouchDB.class);
+        when(mockRemoteDb.changes(null, null, 1000)).then(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                JSONHelper jsonHelper = new JSONHelper();
+                FileReader fr = new FileReader(TestUtils.loadFixture("fixture/empty_docid.json"));
+                return jsonHelper.fromJson(fr, ChangesResult.class);
+            }
+        });
+        when(mockRemoteDb.exists()).thenReturn(true);
+        Collection<String> revs = new ArrayList<String>();
+        revs.add("1-bd42b942b8b672f0289cf3cd1f67044c");
+        when(mockRemoteDb.getRevisions("", revs, new HashSet<String>(), false)).then(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+
+                JSONHelper helper = new JSONHelper();
+                FileReader fileReader = new FileReader(TestUtils.loadFixture
+                        ("fixture/empty_docid_open_revs.json"));
+                List<OpenRevision> openRevs = helper.fromJson(fileReader,
+                        new TypeReference<List<OpenRevision>>() {
+                });
+
+                List<DocumentRevs> documentRevs = new ArrayList<DocumentRevs>();
+
+                for (OpenRevision openRev : openRevs) {
+                    if (openRev instanceof OkOpenRevision) {
+                        documentRevs.add(((OkOpenRevision) openRev).getDocumentRevs());
+                    }
+                }
+
+                return documentRevs;
+            }
+        });
+
+        StrategyListener mockListener = mock(StrategyListener.class);
+        final BasicPullStrategy pullStrategy = new BasicPullStrategy(createPullReplication());
+        pullStrategy.sourceDb = mockRemoteDb;
+        pullStrategy.getEventBus().register(mockListener);
+        pullStrategy.run();
+
+        //should have zero documents
+        Assert.assertEquals(this.datastore.getDocumentCount(),0);
+        //make sure the correct events were fired
+        verify(mockListener).complete(any(ReplicationStrategyCompleted.class));
+        verify(mockListener,never()).error(any(ReplicationStrategyErrored.class));
+
+
+    }
```

Comment:
How difficult would it be to add a test where there are some correct entries (to make sure that the presence of an empty ID parameter doesn't break replication of good change entries)? Right now you're not checking that empty IDs are skipped rather than just the whole thing failing (perhaps silently).